src/frontend/src/views/Upload/components/Metadata/components/StaticTable/components/Row/index.tsx

```diff
@@ -26,21 +34,53 @@ export default memo(function Row({ id, metadata }: Props): JSX.Element {
     publicId,
   } = metadata;
 
+  const validatedCellData: Record<
+    string,
+    React.ReactElement | string | undefined
+  > = {
+    privateId: privateId || "--",
+    collectionDate: collectionDate || "--",
+    collectionLocation:
+      getNameFromCollectionLocation(collectionLocation) || "--",
+    sequencingDate: sequencingDate || "--",
+  };
+
+  if (validationError != null) {
+    Object.entries(validationError).forEach(([key, message]) => {
+      // The validation error for a location is for an id,
+      // which is internal and not something a user would know about.
+      // The only case in which the field does not validate is if the
+      // value is missing entirely.
+      if (key == "collectionLocation") {
+        message = "Required";
+      }
+      validatedCellData[key] = (
```

Comment:
this is probably ok the way it is right now, but how would you feel about moving it into its own component file if it gets any larger?

Reply:
I don't mind